doc/function_types.md

```diff
@@ -1144,3 +1144,66 @@ including ensmallen:
 This can be useful for situations where you know that the checks should be
 ignored.  However, be aware that the code may fail to compile and give more
 confusing and difficult error messages!
+
+
+## Multi-objective functions
+
+A multi-objective function is a set of functions or objectives that are
```

Comment:
I think it makes sense to reference the ArbitraryFunction type, because what this is basically a wrapper, so we should make sure that implements the same API.

Also, looking at the implementation we actually use, `std::tuple<ArbitraryFunctionType...>`, so not sure there is need for `ArbitraryMultiObjectiveFunction` here.

Reply:
Yes true, I added this example to showcase what implementing a multi-objective function (as demonstrated in `ArbitraryMultiObjectiveFunction`) might consist of. Should I remove it?

Reply:
I would remove it, just to avoid any confusion, https://github.com/mlpack/ensmallen/pull/149/files#diff-b0e728716d6532048249f3bcf1562e03R1602 works as an example.